addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiSmartLockHandler.java

```diff
@@ -74,14 +74,23 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     logger.error("Could not refresh Nuki Smart Lock[{}]! Message: {}", nukiId,
                             bridgeLockStateResponse.getMessage());
                 }
-            } else if (channelUID.getId().equals(NukiBindingConstants.CHANNEL_SMARTLOCK_OPEN_CLOSE)
-                    && command instanceof OnOffType) {
-                int lockAction = (command == OnOffType.OFF ? NukiBindingConstants.LOCK_ACTIONS_UNLOCK
-                        : NukiBindingConstants.LOCK_ACTIONS_LOCK);
-                BridgeLockActionResponse bridgeLockActionResponse = new NukiHttpClient(getBridgeConfig())
-                        .getBridgeLockAction(nukiId, lockAction);
-                if (bridgeLockActionResponse.getStatus() != 200) {
-                    logger.error("Could not execute command[{}] on Nuki Smart Lock[{}]", command, nukiId);
+            } else if (command instanceof OnOffType) {
+                int lockAction = -1;
+                if (channelUID.getId().equals(NukiBindingConstants.CHANNEL_SMARTLOCK_OPEN_CLOSE)) {
+                    lockAction = (command == OnOffType.OFF ? NukiBindingConstants.LOCK_ACTIONS_UNLOCK
+                            : NukiBindingConstants.LOCK_ACTIONS_LOCK);
+                } else if (channelUID.getId().equals(NukiBindingConstants.CHANNEL_SMARTLOCK_UNLATCH_CLOSE)) {
+                    lockAction = (command == OnOffType.OFF ? NukiBindingConstants.LOCK_ACTIONS_UNLATCH
+                            : NukiBindingConstants.LOCK_ACTIONS_LOCK);
+                } else {
+                    logger.warn("ChannelUID{} for command{} not implemented!", channelUID, command);
```

Comment:
@martinvw 
I don't get it or maybe I get it wrong but:
I'm assigning a negative value to lockAction on line 78
Then from line 79 - 87 I'm checking the channel and if it's a valid channel I'm changing the lockAction to the appropriate lock action value
Then on line 88 I'm checking this lockAction != -1 and just then I execute the lock command

Also, I'm don't know what you mean with "if you return"?

Thx